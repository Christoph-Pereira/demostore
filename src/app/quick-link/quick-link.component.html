<div class="quickLinkDiv">

    <h1>{{headline}}</h1>

    <div *ngIf="url?.indexOf('http') === 0; else elseBlock">
        <a [href]="url" [target]="newTab ? '_blank' : '_self'" rel="noopener noreferrer">
            {{text}}
        </a>
    </div>
    <ng-template #elseBlock>
        <a [routerLink]="url" (click)="changeHeaderBackground()">
            {{text}}
        </a>
    </ng-template>
</div>