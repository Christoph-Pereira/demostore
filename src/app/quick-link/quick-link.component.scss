.quickLinkDiv {
    margin: auto;

    a {
        text-decoration: none;
        color: blue;
    }
}