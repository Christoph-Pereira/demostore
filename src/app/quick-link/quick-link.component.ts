import { Component, Input, OnInit } from '@angular/core';
import { HeaderColorService } from '../services/header-color.service';

@Component({
  selector: 'app-quick-link',
  templateUrl: './quick-link.component.html',
  styleUrls: ['./quick-link.component.scss']
})
export class QuickLinkComponent {

  @Input() public headline: string | undefined;
  @Input() public text: string | undefined;
  @Input() public url: string | undefined;
  @Input() public newTab: boolean | undefined;

  constructor(private headerColor: HeaderColorService) {
   }

  changeHeaderBackground() {
    this.headerColor.setColorSubject("black");
  }

}
