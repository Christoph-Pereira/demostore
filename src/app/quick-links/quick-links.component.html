<div class="quickLinks">
    <div *ngFor="let quickLink of quickLinks">
        <app-quick-link [headline]="quickLink.headline" [text]="quickLink.text" [url]="quickLink.url" [newTab]="quickLink.newTab"></app-quick-link>
    </div>
</div>