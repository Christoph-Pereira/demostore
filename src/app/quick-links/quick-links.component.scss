.quickLinks {
    display: grid;
    grid-template-columns: auto auto auto;
    padding: 3rem 20rem;
}