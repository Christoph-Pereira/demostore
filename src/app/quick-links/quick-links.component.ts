import { Component, OnInit } from '@angular/core';
import { ProductService } from '../api/products.service';
import { UserServiceService } from '../services/user-service.service';
import { Observable } from 'rxjs';

interface quickLink {
  headline: string,
  text: string,
  url: string,
  newTab: boolean
}
@Component({
  selector: 'app-quick-links',
  templateUrl: './quick-links.component.html',
  styleUrls: ['./quick-links.component.scss']
})

export class QuickLinksComponent implements OnInit {

  public quickLinks: quickLink[] | undefined;
  public randomProductTitleText: Observable<string> = this.productService.getRandomProductTitle();
  public getFavoriteProduct: Observable<string> = this.userService.getFavoriteProduct();
  public doesCartContainFavoriteProduct: Observable<boolean> = this.userService.doesCartContainFavoriteProduct();

  constructor(private productService: ProductService, private userService: UserServiceService) { }

  ngOnInit(): void {
    this.quickLinks = [
      {
        "headline": "QuickLinksLink01",
        "text": "Example QuickLink >",
        "url": "/configurator",
        "newTab": false
      },
      {
        "headline": "QuickLinksLink02",
        "text": "Example QuickLink >",
        "url": "/search/need",
        "newTab": true
      },
      {
        "headline": "QuickLinksLink03",
        "text": "Example QuickLink >",
        "url": "https://angular.io/",
        "newTab": false
      }
    ];
  }

}
