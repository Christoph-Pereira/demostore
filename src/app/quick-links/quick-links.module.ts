import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { QuickLinkComponent } from '../quick-link/quick-link.component';
import { QuickLinksComponent } from './quick-links.component';
import { RouterModule } from '@angular/router';



@NgModule({
  declarations: [QuickLinkComponent, QuickLinksComponent],
  imports: [
    CommonModule,
    RouterModule
  ],
  exports: [QuickLinksComponent]
})
export class QuickLinksModule { }
