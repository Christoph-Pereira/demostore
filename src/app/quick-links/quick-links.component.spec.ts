import { ComponentFixture, TestBed } from '@angular/core/testing';

import { QuickLinksComponent } from './quick-links.component';
import { QuickLinkComponent } from '../quick-link/quick-link.component';
import { MockComponent } from 'ng-mocks';

describe('QuickLinksComponent', () => {
  let component: QuickLinksComponent;
  let fixture: ComponentFixture<QuickLinksComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [ QuickLinksComponent, MockComponent(QuickLinkComponent) ]
    })
    .compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(QuickLinksComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should contain quickLinks', () => {
    expect(component.quickLinks).toBeTruthy();
  });

  it('should contain quicklinkslink01', () => {
    const quickLinks = component.quickLinks;
    const exampleQuickLink = {
      "headline": "QuickLinksLink01",
      "text": "Example QuickLink >",
      "url": "/configurator",
      "newTab": false
    }
    expect(quickLinks).toContainEqual(exampleQuickLink);
  });
  
  it('should match snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });
});
