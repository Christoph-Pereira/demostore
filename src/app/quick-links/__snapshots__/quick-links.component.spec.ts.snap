// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuickLinksComponent should match snapshot 1`] = `
<app-quick-links
  __ngContext__={[Function LRootView]}
  quickLinks={[Function Array]}
>
  <div
    class="quickLinks"
  >
    <div>
      <app-quick-link />
    </div>
    <div>
      <app-quick-link />
    </div>
    <div>
      <app-quick-link />
    </div>
  </div>
</app-quick-links>
`;
